types/react-redux/index.d.ts

```diff
@@ -445,3 +447,240 @@ export const ReactReduxContext: Context<ReactReduxContextValue>;
  * multiple actions dispatched outside of React only result in a single render update.
  */
 export function batch(cb: () => void): void;
+
+// tslint:disable:no-unnecessary-generics
+
+/**
+ * A hook to bind action creators to the redux store's `dispatch` function
+ * similar to how redux's `bindActionCreators` works.
+ *
+ * Supports passing a single action creator, an array/tuple of action
+ * creators, or an object of action creators.
+ *
+ * Any arguments passed to the created callbacks are passed through to
+ * your functions.
+ *
+ * This hook takes a dependencies array as an optional second argument,
+ * which when passed ensures referential stability of the created callbacks.
+ *
+ * @param actions the action creators to bind
+ * @param deps (optional) dependencies array to control referential stability
+ *
+ * @returns callback(s) bound to store's `dispatch` function
+ *
+ * @example
+ *
+ * import React from 'react'
+ * import { useActions } from 'react-redux'
+ *
+ * const increaseCounter = ({ amount }) => ({
+ *   type: 'increase-counter',
+ *   amount,
+ * })
+ *
+ * export const CounterComponent = ({ value }) => {
+ *   // supports passing an object of action creators
+ *   const { increaseCounterByOne, increaseCounterByTwo } = useActions({
+ *     increaseCounterByOne: () => increaseCounter(1),
+ *     increaseCounterByTwo: () => increaseCounter(2),
+ *   }, [])
+ *
+ *   // supports passing an array/tuple of action creators
+ *   const [increaseCounterByThree, increaseCounterByFour] = useActions([
+ *     () => increaseCounter(3),
+ *     () => increaseCounter(4),
+ *   ], [])
+ *
+ *   // supports passing a single action creator
+ *   const increaseCounterBy5 = useActions(() => increaseCounter(5), [])
+ *
+ *   // passes through any arguments to the callback
+ *   const increaseCounterByX = useActions(x => increaseCounter(x), [])
+ *
+ *   return (
+ *     <div>
+ *       <span>{value}</span>
+ *       <button onClick={increaseCounterByOne}>Increase counter by 1</button>
+ *     </div>
+ *   )
+ * }
+ */
+export function useActions<
+    T extends [ActionCreator<A>, ActionCreator<A>, ActionCreator<A>, ActionCreator<A>, ActionCreator<A>, ActionCreator<A>, ActionCreator<A>, ActionCreator<A>, ActionCreator<A>],
```

Comment:
I'm still running tests locally, but we can likely replace all these overloads with a single `T extends ReadonlyArray<readonly ActionCreator<A>>`. That prevents the action creator tuple from getting widened to an array.

Reply:
Typescript couldn't infer the type of data in every index without tuples **for now**. 

```ts
// With action creator tuples
useAction([a, b]) //=> [A, B]

// Without action creator tuples
useAction([a. b]) //=> (A | B)[]
```

Reply:
Looks like it only works properly on TypeScript 3.4 and above. Too bad, but it's a good option for a couple months from now...

Reply:
Is it possible to release a major version that has a peer dependency on typescript@^3.4 now?

Reply:
The compatibility with TypeScript 3.0 is exactly the reason why I typed it this way. I think these typings are just bearable enough that I wouldn't want to bump the peer dependency just for this. In a couple of months it's an easy fix.